mongo/errors.go

```diff
@@ -71,6 +73,56 @@ func replaceErrors(err error) error {
 	return err
 }
 
+// IsDuplicateKeyError returns true if err is a duplicate key error
+func IsDuplicateKeyError(err error) bool {
+	if e, ok := err.(ServerError); ok {
+		return e.HasErrorCode(11000) || e.HasErrorCode(11001) || e.HasErrorCode(12582) ||
+			e.HasErrorCodeWithMessage(16460, " E11000 ")
+	}
+	return false
+}
+
+// IsTimeout returns true if err is from a timeout
+func IsTimeout(err error) bool {
+	//timeout error labels
+	if se, ok := err.(ServerError); ok {
```

Comment:
Do we need to check wether `err` wraps a `ServerError`?

If a user wraps a `ServerError` should they be able to use these helpers directly?

```
func DoPing(...) error {
	err := client.Ping(ctx, nil)
	if err != nil {
		return fmt.Errorf("db error: %w", err)
	}
	return nil
}

err := DoPing (...)
if mongo.IsTimeout (err) {
   ...
}
```